### PR TITLE
Update Intercom Dev Doc for Intercom Source v2

### PR DESCRIPTION
**Pull request by ampcynthli**
> You can see the user property mapping preference you selected from the data source detail page for Intercom.
Looking to improve the wording here. The page I'm referring to is this: https://analytics.amplitude.com/amplitude/connections/project/187520/sources/detail/bGlicmFyeTphZGFwdGVyL2ludGVyY29t

Here is a lucid diagram of the new flow: https://lucid.app/lucidchart/a20cb365-6aab-450b-9f26-eec165c6507e/edit?viewport_loc=-139%2C-53%2C2269%2C1245%2CTqnafomwO3dB&invitationId=inv_836b9c96-c5ba-4973-b29e-7c39c87996d0

Intercom Source v2 Setup - Front end UX

# Amplitude Developer Docs PR

## Description

- Updated wording to reflect properties of new Intercom source integration
  - users no longer require a valid intercom ID that maps to Amplitude users. Instead, a choice can be made to map from Intercom users' user ID or email to Amplitude User IDs.
- Updated set up guide to better illustrate intercom V2 source set up steps

## Deadline

When do these changes need to be live on the site?
Once the new UI is deployed to production.
- Estimation: December 14

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [X] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [X] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [X] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp
@Brandoncharleskhoo1

### CODE REVIEW

File: docs/data/sources/intercom.md [hunk 1]
```diff
@@ -17,7 +17,9 @@ Sync Intercom event data to Amplitude so that you can better engage your users b
 ## Considerations
 
 - You need both an Amplitude and an Intercom account.
-- The user IDs in Intercom and Amplitude must be the same for each user in order for this integration to work. If the Intercom user ID doesn't match Amplitude's user ID for the same user, Amplitude interprets the user as new. If there is no Intercom user ID at all, Amplitude drops the events.
+- You may select either Intercom users' User ID property or their email property to map to Amplitude users' User ID.
+- Users in Amplitude and Intercom must either have the same user IDs or emails for this integration to work. You can select which property to match on; but, if the property does not match for the same user between Amplitude and Intercom, Amplitude interprets the user as new.
```
Comment (caseyamp) [inline docs/data/sources/intercom.md:21]:
```suggestion
- Users in Amplitude and Intercom must either have the same user ID or email for this integration to work. You can choose which property to match on. If the property doesn't match for the same user between Amplitude and Intercom, Amplitude interprets the user as new. 
```

**Review by caseyamp — COMMENTED** [review]
@ampcynthli -- just a few minor suggestions. When do you want this merged?

**Comment by ampcynthli** [comment]
> @ampcynthli -- just a few minor suggestions. When do you want this merged?

Thanks for the suggestion, Casey. We want to merge this once Intercom source V2 is released and available to customers, but we don't have a set date for this just yet (but it's by the end of Q4 for sure). I'll let you know when we'd like this to be merged if that's fine with you?

**Comment by caseyamp** [comment]
@ampcynthli of course! Just let me know when y'all are ready to ship. I don't have any PTO planned other than regular US holidays through the end of the year, so just ping me when ready.

**Comment by ampcynthli** [comment]
@caseyamp Hey, this should be ready to ship. We've enabled the flag for all customers, and the new UI is live.

**Review by amplitude-dev-docs — APPROVED** [review]

**Pull request merged**